storage/leveldb/leveldb.go

```diff
@@ -94,13 +95,7 @@ func (s *DB) batchTimeoutHandle() {
 	}
 }
 
-// Put adds the value to the (key, val) storage medium
-func (s *DB) Put(key, val []byte) error {
-	err := s.batch.Put(key, val)
-	if err != nil {
-		return err
-	}
-
+func (s *DB) updateBatchWithIncrement() error {
 	s.mutBatch.Lock()
```

Comment:
I think this mutex is no longer required, the batch struct just became concurrent safe and we are reusing the same batch

Reply:
it is required as it also protects the current size of the batch